app/scripts/controllers/console.js

```diff
@@ -128,6 +205,67 @@ const crate_console = angular.module('console', ['sql', 'datatypechecks', 'stats
           scrollTop();
         };
 
+        $scope.export = function(format){
+          var rows = $scope.rows;
+          var cols = $scope.resultHeaders;
+          var col_types = $scope.resultHeaderTypes;
+
+          switch (format.id) {
+            case 'json':
+              var objs = queryResultToObjects({ rows: rows }, cols);
+              ExportUtils.saveFile('export.json', 'application/json', JSON.stringify(objs));
+              break;
+            case 'csv':
+              var separator = ',';
+              var linebreak = '\r\n';
+              var output = '';
+              // Header
+              output += cols.join(separator) + linebreak;
+              // Data
+              rows.forEach((row, index) => {
+                var last = index === rows.length - 1;
+    
+                row.forEach((column, index) => {
+                  var last = index === row.length - 1;
+                  var type = getDataType(column, col_types[index]);
+    
+                  switch (type) {
+                    case 'object':
+                    case 'array':
+                      // Stringify twice to escape the output
+                      output += JSON.stringify((JSON.stringify(column)));
```

Comment:
I don't understand why stringify is applied twice ? 🤔 Maybe you can add a test for this case ?

Reply:
okay I see, then It would be great if you test the export functionality for different cases to see that it matches the expected csv or json output